From 99ee802c8a3d1b1a240ad758741f017730ac3799 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Mon, 13 Sep 2021 14:59:30 +0300
Subject: [PATCH] [SW-55796] gaudi2: increase hbm bar size to 64gb

BAR size for HBM is set to 64GB instead of default 32GB.
This is required for host for access of all available
space, so that host won't need to shift small BAR window
for every access which is outside the window.
Therefore, this method is more efficient and more secured.

Change-Id: I81654bc0a3eff00b0a3b5f3849d58d99b7bd9a43
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index db549b4..3a7673b 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -9,12 +9,26 @@
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
 #include <address_space.h>
+#include <pcie_aux_regs.h>
+#include <pcie_dbi_regs.h>
+#include <pcie_aux_masks.h>
 
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
 #include <pci.h>
 
+#define HBM_BAR_PF_NUM (1 << 4)
+
+__weak void preboot_update_bar_sizes(void)
+{
+	/* Set HBM BAR size to 64GB */
+	regmap_writel(HBM_BAR_PF_NUM << PCIE_AUX_DBI_PF_NUM_SHIFT, mmPCIE_AUX_DBI);
+	regmap_writel(0xFFFFFFFF, mmPCIE_DBI_BAR4_REG);
+	regmap_writel(0x0000000F, mmPCIE_DBI_BAR5_REG);
+	regmap_writel(0x00000000, mmPCIE_AUX_DBI);
+}
+
 static struct hl_pll_cfg gaudi2_pll_cfg[] = {
 	{
 		.name = "DCORE0_IF",
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 5a8d412..6c72bfa 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -420,6 +420,9 @@
 	regmap_writel(0x0004A07D, mmPCIE_PHY_CR_PARA_CMD);
 	udelay(20);
 
+	/* Set HBM BAR size */
+	preboot_update_bar_sizes();
+
 	/* PCIe disable LTR */
 	preboot_disable_pcie_ltr();
 
diff --git a/arch/gaudi2/common/onix.c b/arch/gaudi2/common/onix.c
index f6289e6..c4ebe18 100644
--- a/arch/gaudi2/common/onix.c
+++ b/arch/gaudi2/common/onix.c
@@ -44,3 +44,5 @@
 void hl_dev_ts_init(void) {}
 
 void hl_ts_shutdown(void) {}
+
+void preboot_update_bar_sizes(void) {}
diff --git a/arch/gaudi2/common/pldm.c b/arch/gaudi2/common/pldm.c
index 672da66..2252c2e 100644
--- a/arch/gaudi2/common/pldm.c
+++ b/arch/gaudi2/common/pldm.c
@@ -63,3 +63,5 @@
 }
 
 void preboot_init_pcie_phy(void) {}
+
+void preboot_update_bar_sizes(void) {}
diff --git a/include/preboot.h b/include/preboot.h
index b69a9ae..4c1981c 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -227,6 +227,7 @@
 void preboot_set_fw_cfg_skip(u8 skip);
 void preboot_set_binning_conf(struct hl_binning_masks *masks);
 void preboot_finalize_uboot_override(void);
+void preboot_update_bar_sizes(void);
 bool hl_fw_cfg_skip_is_supported(void);
 bool hl_binning_conf_is_supported(void);
 
